File: src/main/java/org/funivan/lologa/algo/ga/genome/value/LoggableValueInterface.java
package org.funivan.lologa.algo.ga.genome.value;

import org.funivan.lologa.tiles.TilesInterface;

public class LoggableValueInterface implements ValueInterface {
    private final boolean show;
    private final ValueInterface origin;

    public LoggableValueInterface(boolean show, ValueInterface origin) {
        this.show = show;
        this.origin = origin;
    }

    @Override
    public String type() {
        return this.origin.type();
    }

    @Override
    public Double value(TilesInterface original, TilesInterface tiles) {
        Double v = this.origin.value(original, tiles);
        if (this.show) {
            System.out.println(this.getClass() + " = " + v);
        }
        return v;
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/value/LoggableValueInterface.kt
package org.funivan.lologa.algo.ga.genome.value

import org.funivan.lologa.tiles.TilesInterface

class LoggableValueInterface(private val show: Boolean, private val origin: ScoreInterface) : ScoreInterface {

    override fun type(): String {
        return this.origin.type()
    }

    override fun value(original: TilesInterface, tiles: TilesInterface): Double {
        val result = this.origin.value(original, tiles)
        if (this.show) {
            println(this.javaClass.toString() + " = " + result)
        }
        return result
    }
}
